src/core/renderer.js

```diff
@@ -165,11 +165,35 @@ class Renderer {
 
         viewportExpressions.forEach(expr => expr._resetViewportAgg());
 
+        const viewportExpressionsLength = viewportExpressions.length;
+
         // Avoid acumulating the same feature multiple times keeping a set of processed features (same feature can belong to multiple dataframes).
         const processedFeaturesIDs = new Set();
 
+
+        this.aspect= this.gl.canvas.width / this.gl.canvas.height;
```

Comment:
instead mutating this, I`d create a constant and i´d use this constant as a parameter in `_isFeatureInViewport``

```js
constant ASPECT_RATIO = this.gl.canvas.width / this.gl.canvas.height;

// ...

 if (!this._isFeatureInViewport(dataframe, i, ASPECT_RATIO)) {
// ...